tasks/install_shell.sh

```diff
@@ -112,6 +112,7 @@ if [ -f "$PT__installdir/facts/tasks/bash.sh" ]; then
       "10.11") platform_version="10.11";;
       "10.12") platform_version="10.12";;
       "10.13") platform_version="10.13";;
+      "10.14") platform_version="10.14";;
```

Comment:
@donoghuc osx-10.14 was shipped.

Reply:
For some reason I'm not seeing it at https://downloads.puppetlabs.com/mac/10.14/PC1/x86_64/

Reply:
It was shipped for the puppet5 and puppet6 collections (stream) -- PC1 (in that url) is the name of the collection that contains puppet 4 (for PE 2016.4.x), which is far into the end of its life. When we add new platforms, we ship them for currently supported collections of puppet only -- this means there aren't any puppet 4 packages for 10.14.

Reply:
Sorry @caseywilliams That was not the link I meant to post (I was poking around and copied the wrong URL). With the change in this PR we try to get (http://downloads.puppetlabs.com/mac/puppet/10.14/x86_64/) not PC1. 

This could be another example of how the `install_shell task` implementation needs a refresh. We set `collection` to `puppet` by default. We build the url with:
https://github.com/puppetlabs/puppetlabs-puppet_agent/blob/3809e05dbf86913bee07c06159313a9e01b85379/tasks/install_shell.sh#L479
You can change that with a parameter to the task. What `collection` should we expect the 10.14 dmg to be in? It seems like the options would be `puppet`, `puppet5' and `puppet6`. 
at `downloads.puppetlabs.com/mac/{puppet,puppet5,puppet6}/10.14/x86_64/`

Reply:
Ah, I see -- those collection names (puppet, puppet5, and puppet6) are correct, but my comment was totally misleading -- the packages aren't in those collections on downloads.puppet.com because we haven't shipped an official release that includes 10.14 yet (we've only shipped them to nightlies, here: http://nightlies.puppet.com/downloads/mac/)

Reply:
Perhaps the task addition should come in a separate PR, later, after the officially released packages become available. We'll also need to be careful how we enable them -- the 6.3.0 release is coming up and will have 10.14 packages, but the next release in the puppet5 series won't be until later.